pkg/api/norman/customization/cluster/validator.go

```diff
@@ -342,6 +346,127 @@ func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec
 
 }
 
+func (v *Validator) validateAKSConfig(request *types.APIContext, cluster map[string]interface{}, clusterSpec *v32.ClusterSpec) error {
+	aksConfig, ok := cluster["aksConfig"].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	var prevCluster *v3.Cluster
+
+	if request.Method == http.MethodPut {
+		var err error
+		prevCluster, err = v.ClusterLister.Get("", request.ID)
+		if err != nil {
+			return err
+		}
+	}
+
+	createFromImport := request.Method == http.MethodPost && aksConfig["imported"] == true
+
+	if !createFromImport {
+		if err := validateAKSKubernetesVersion(clusterSpec, prevCluster); err != nil {
+			return err
+		}
+		if err := validateAKSNodePools(clusterSpec); err != nil {
+			return err
+		}
+		if err := validateAKSAccess(request, aksConfig, prevCluster); err != nil {
+			return err
+		}
+	}
+
+	if request.Method != http.MethodPost {
+		return nil
+	}
```

Comment:
This could be removed if there is no other create-only validation to do.

Reply:
Removed